compiler/lib/eval.ml

```diff
@@ -359,14 +422,17 @@ let eval_instr info ((x, loc) as i) =
                 ( x
                 , Prim
                     ( prim
-                    , List.map2 prim_args prim_args' ~f:(fun arg c ->
-                          match c with
-                          | Some ((Int _ | Float _ | NativeString _) as c) -> Pc c
-                          | Some (String _ as c) when Config.Flag.use_js_string () -> Pc c
-                          | Some _
+                    , List.map2 prim_args prim_args' ~f:(fun arg (c : constant option) ->
+                          match c, Config.target () with
+                          | ( Some ((Int _ | Int32 _ | NativeInt _ | NativeString _) as c)
```

Comment:
Do we really want to have Int32 and NativeInt in this branch ?

Reply:
Fixed